File: aoc2018/01.rs
use std::fs;
use std::io::{self, BufRead};

fn main() -> io::Result<()> {
    // Open the file in read-only mode with buffer.
    let file = fs::File::open("01.txt")?;
    let reader = io::BufReader::new(file);

    // Read the file line by line.
    let mut numbers = Vec::new();
    for line in reader.lines() {
        // Parse each line as a number.
        match line?.parse::<i32>() {
            Ok(n) => numbers.push(n),
            Err(e) => {
                println!("Error: {}", e)
            }
        }
    }

    // Do something with the numbers.
    let mut freq = 0;
    for number in numbers {
        freq += number;
    }
    println!("freq: {}", freq);

    Ok(())
}

File: aoc2015/01.py
IN = open("01.txt").read()

print(IN.count("(") - IN.count(")"))

floor = 0
for idx, c in enumerate(IN):
    floor += {"(": 1, ")": -1}[c]
    if floor < 0:
        print(idx + 1)
        break
